raiden/network/proxies/user_deposit.py

```diff
@@ -103,55 +107,61 @@ def deposit(
             "total_deposit": total_deposit,
         }
 
-        checking_block = self.client.get_checking_block()
         error_prefix = "Call to deposit will fail"
 
         with self.deposit_lock:
-            amount_to_deposit, log_details = self._deposit_preconditions(
-                total_deposit=total_deposit,
-                beneficiary=beneficiary,
-                token=token,
-                block_identifier=block_identifier,
-            )
-
-            with log_transaction(log, "deposit", log_details):
-                gas_limit = self.proxy.estimate_gas(
-                    checking_block, "deposit", to_checksum_address(beneficiary), total_deposit
+            # check preconditions
+            try:
+                previous_total_deposit = self.get_total_deposit(
+                    address=beneficiary, block_identifier=given_block_identifier
                 )
-
-                if gas_limit:
-                    error_prefix = "Call to deposit failed"
-                    gas_limit = safe_gas_limit(gas_limit)
-                    log_details["gas_limit"] = gas_limit
-
-                    transaction_hash = self.proxy.transact(
-                        "deposit", gas_limit, to_checksum_address(beneficiary), total_deposit
+                current_balance = token.balance_of(
+                    address=self.node_address, block_identifier=given_block_identifier
+                )
+            except ValueError:
+                # If 'given_block_identifier' has been pruned, we cannot perform the
+                # precondition checks but must still set the amount_to_deposit to a
+                # reasonable value.
+                previous_total_deposit = self.get_total_deposit(
+                    address=beneficiary, block_identifier=self.client.get_checking_block()
+                )
+                amount_to_deposit = TokenAmount(total_deposit - previous_total_deposit)
+            except BadFunctionCallOutput:
+                raise RaidenUnrecoverableError(
```

Comment:
Why was it hard to use the exact same line used everywhere else?
```
raise_on_call_returned_empty(given_block_identifier)
```

Reply:
`raise_on_call_returned_empty` is in the `token_network` module. I didn't want to import it/move it to `utils` when all it does is raise an exception.